cpp/arcticdb/entity/performance_tracing.cpp

```diff
@@ -13,6 +13,8 @@
 #include <arcticdb/util/preprocess.hpp>
 #include <arcticdb/util/pb_util.hpp>
 
+#if defined(USE_REMOTERY)
```

Comment:
Here are the occurrences of the class outside of this file:

```
cpp/arcticdb/python/python_module.cpp:234:    py::class_<RemoteryInstance, std::shared_ptr<RemoteryInstance>>(remotery, "Instance");
cpp/arcticdb/util/global_lifetimes.cpp:30:    RemoteryInstance::destroy_instance();
cpp/arcticdb/version/local_versioned_engine.cpp:35:    auto temp = RemoteryInstance::instance();
```

What would you recommend doing instead? To conditionally define the class to implement its methods when `USE_REMOTERY` is defined?

Reply:
I was thinking of putting the `#if` in the body of the `RemoteryInstance` methods and the `rmt_` field, so they do nothing if Remotery is not enabled, but the class still exists.

That way this PR can be reduced to a couple of files.

Reply:
I thought of this alternative but eventually did not choose it as I thought this could mislead readers.

The current solution is touching a few files but I think is a bit more explicit, and does not create unusable objects.

What do you think?

Reply:
As long as you have considered that alternative and there's written record of the deliberation (here), I am fine with either solution.

Reply:
I think that I prefer this alternative. I have also removed the useless noise with c50cde9. 

In the long term, I think it would be valuable to move custom tools' setup (like Remotery) entirely to dedicated branches rebased on `master` to ease the CMake setup. What do you think?

For now, I let you approve and merge if this looks good to you to make the decision clear and consensual.